components/data-bridge/org.wso2.carbon.databridge.agent/src/main/java/org/wso2/carbon/databridge/agent/endpoint/DataEndpointGroup.java

```diff
@@ -135,9 +135,14 @@ private void tryPut(Event event, long timeoutMS) throws EventQueueFullException
                     this.ringBuffer.publish(sequence);
                     break;
                 } catch (InsufficientCapacityException ex) {
-                    if (stopTime > System.currentTimeMillis()) {
+                    if (stopTime <= System.currentTimeMillis()) {
                         throw new EventQueueFullException("Cannot send events because the event queue is full", ex);
                     }
+                    try {
+                        Thread.sleep(1);
+                    } catch (InterruptedException e) {
+
```

Comment:
Sajith, we don't need a thread sleep there. We can continousely tryPut to queue until it's successful. hence please remove the thread sleep part and send a pull request.